src/main/java/com/powsybl/openloadflow/network/impl/LfBranchImpl.java

```diff
@@ -35,39 +35,95 @@
 
     private Evaluable q2 = NAN;
 
-    protected LfBranchImpl(LfBus bus1, LfBus bus2, PiModel piModel, Branch branch) {
+    protected LfBranchImpl(LfBus bus1, LfBus bus2, PiModel piModel, PhaseControl phaseControl, Branch branch) {
         super(bus1, bus2, piModel);
+        this.phaseControl = phaseControl;
         this.branch = branch;
     }
 
+    private static LfBranchImpl createLine(Line line, LfBus bus1, LfBus bus2, double zb) {
+        PiModel piModel = new SimplePiModel()
+                .setR(line.getR() / zb)
+                .setX(line.getX() / zb)
+                .setG1(line.getG1() * zb)
+                .setG2(line.getG2() * zb)
+                .setB1(line.getB1() * zb)
+                .setB2(line.getB2() * zb);
+
+        return new LfBranchImpl(bus1, bus2, piModel, null, line);
+    }
+
+    private static LfBranchImpl createTransformer(TwoWindingsTransformer twt, LfBus bus1, LfBus bus2, double nominalV1,
+                                                  double nominalV2, double zb) {
+        PiModel piModel;
+        PhaseControl phaseControl = null;
+
+        PhaseTapChanger ptc = twt.getPhaseTapChanger();
```

Comment:
[Memo] We have to put an issue to say that phase control on T3W will not be supported in the first version.

Reply:
t3w is not handled here it is just for t2w. t3w code is in LfLegBranch